crates/jarscanner/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+use crate::errors::FileNameError;
+use serde::{Deserialize, Serialize};
+
+use std::collections::HashMap;
+use std::error::Error;
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::PathBuf;
+use zip::read::ZipArchive;
+
+pub mod errors;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default, Eq)]
+struct DataBlock {
+    entity_path: String,
+    defs: Vec<String>,
+    refs: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default, Eq)]
+pub struct TargetDescriptor {
+    label_or_repo_path: String,
+    data_blocks: Vec<DataBlock>,
+}
+
+fn non_anon(file_name: &str) -> bool {
```

Comment:
we should add a comment here that this is to filter scala generated classes that users should never reference.